src/app/user/user.service.ts

```diff
@@ -0,0 +1,25 @@
+import { inject, injectable } from "tsyringe";
+import { Repository } from "../repository";
+import { User } from "./user.model";
+
+@injectable()
+export class UserService {
+  constructor(
+    @inject("UserRepository") private userRepository: Repository<User>
+  ) {}
+
+  getUserStatus = async (userId: string) => {
+    const user = await this.userRepository.findOne({ id: userId });
+
+    const totalNumberOfWalls: number = user.houses.reduce(
+      (accumulator, current) => {
+        return accumulator + current.walls;
```

Comment:
The issue exists here.
Because the houses property on the user object has types ObjectId or House array, typescript complains about indexing into the union because ObjectId does not contain property "walls"